### PR TITLE
9178 chapter 31 e2e tests

### PR DESCRIPTION
**Pull request by rianfowler**
Probably done but I had questions about a couple of things that might require updates.

### CODE REVIEW

File: src/js/vre/chapter31/config/form.js [hunk 1]
```diff
@@ -176,23 +176,13 @@ const formConfig = {
...
+            employer: _.merge({
               'ui:title': 'Employer name',
               'ui:required': (formData) => formData['view:isWorking'],
-              ...expandIfWorking
-            },
-            jobDuties: {
-              'ui:title': 'Job duties',
-              ...expandIfWorking
-            },
-            monthlyIncome: {
```
Comment (rianfowler) [inline src/js/vre/chapter31/config/form.js]: This doesn't assign the correct class to the element. Just refactored everything to use `_.merge`

File: test/e2e/vre-ch31-helpers.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
+const mock = require('./mock-helpers');
+
+function initApplicationSubmitMock() {
+  mock(null, {
```
Comment (rianfowler) [inline test/e2e/vre-ch31-helpers.js]: not sure how this works- this is creating an mock end point that provides the `value` as the response?

File: test/e2e/vre-ch31-helpers.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    .fill('input[name=root_disabilities]', data.disabilities)
+    .fill('input[name=root_vaRecordsOffice]', data.vaRecordsOffice)
+    .selectRadio('root_view:inHospital', data['view:inHospital'])
+    .fill('input[name=root_view\\:hospital_hospitalName]', data.hospitalName)
+    .fillAddress('root_view:hospital_hospitalAddress', data.hospitalAddress);
+}
+
+function completeDischargeDocumentUpload(client) {
+  client
+    .setValue('input#root_dischargeDocuments', require('path').resolve(`${__dirname}/VA40-10007.pdf`));
```
Comment (rianfowler) [inline test/e2e/vre-ch31-helpers.js]: I copied this from vic-v2 but I don't think it's actually doing anything.

File: test/e2e/vre-ch31-helpers.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+function completeDisabilityInformation(client, data) {
+
+  client
+    .selectDropdown('root_disabilityRating', data.disabilityRating)
+    .fill('input[name=root_disabilities]', data.disabilities)
+    .fill('input[name=root_vaRecordsOffice]', data.vaRecordsOffice)
+    .selectRadio('root_view:inHospital', data['view:inHospital'])
+    .fill('input[name=root_view\\:hospital_hospitalName]', data.hospitalName)
```
Comment (rianfowler) [inline test/e2e/vre-ch31-helpers.js]: `\\` 🤷‍♂️

File: test/vre/chapter31/00-all-fields.e2e.spec.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+    E2eHelpers.overrideVetsGovApi(client);
+    E2eHelpers.overrideSmoothScrolling(client);
+    E2eHelpers.expectNavigateAwayFrom(client, '/introduction');
+
+    // Veteran information page
+    client.waitForElementPresent('input[name="root_veteranFullName_first"]', Timeouts.normal);
+    client.assert.cssClassPresent('.progress-bar-segmented div.progress-segment:nth-child(1)', 'progress-segment-complete');
+    PageHelpers.completeVeteranInformation(client, testData.data);
+    client.axeCheck('.main')
```
Comment (rianfowler) [inline test/vre/chapter31/00-all-fields.e2e.spec.js]: What is this doing?

File: test/e2e/vre-ch31-helpers.js (hunk 2 above)
Comment (jbalboni) [reply test/e2e/vre-ch31-helpers.js]: That is exactly it. It's confusing to read on first pass.

File: test/e2e/vre-ch31-helpers.js (hunk 4 above)
Comment (jbalboni) [reply test/e2e/vre-ch31-helpers.js]: We usually don't need that in e2e tests. Maybe by putting " around the attribute value?

File: test/e2e/vre-ch31-helpers.js (hunk 3 above)
Comment (jbalboni) [reply test/e2e/vre-ch31-helpers.js]: The document upload e2e stuff doesn't work very well on Jenkins/Electron, so probably not.

File: test/vre/chapter31/00-all-fields.e2e.spec.js (hunk 5 above)
Comment (jbalboni) [reply test/vre/chapter31/00-all-fields.e2e.spec.js]: It's running the aXe accessibility checker on the DOM inside the main element.

File: test/vre/chapter31/schema/maximal-test.json [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    "veteranFullName": {
+      "first": "first",
+      "middle": "middle",
+      "last": "last",
+      "suffix": "Sr."
+    },
+    "veteranDateOfBirth": "1970-03-03",
+    "veteranSocialSecurityNumber": "424242424",
+    "veteranVaFileNumber": "3423434",
+    "view:inHospital": "Y",
```
Comment (jbalboni) [inline test/vre/chapter31/schema/maximal-test.json]: I think these flags are booleans?

**Review by jbalboni — APPROVED** [review]
Looks good to me.

File: test/vre/chapter31/schema/maximal-test.json (hunk 6 above)
Comment (rianfowler) [reply test/vre/chapter31/schema/maximal-test.json]: Will update

Comment (rianfowler) [reply test/vre/chapter31/schema/maximal-test.json]: I see what the problem is- I was using `selectRadio` instead of `selectYesNo`

**Pull request merged**